internal/controller/utils.go

```diff
@@ -469,3 +474,95 @@ func healthCheck(sts *appsv1.StatefulSet, lg klog.Logger) (*clientv3.MemberListR
 
 	return memberlistResp, healthInfos, nil
 }
+
+func (r *EtcdClusterReconciler) getStatefulSetPods(sts *appsv1.StatefulSet, ctx context.Context) (*corev1.PodList, error) {
+	podList := corev1.PodList{}
+	err := r.Client.List(ctx, &podList, client.InNamespace(sts.Namespace), client.MatchingLabels(sts.Spec.Selector.MatchLabels))
+	if err != nil {
+		return nil, err
+	}
+	return &podList, nil
+}
```

Comment:
Is it possible to reuse `getStatefulSet` instead of adding a new method?

https://github.com/etcd-io/etcd-operator/blob/2c9366ae2f23f02b10ceb24072a5b3ffe5c6f0e2/internal/controller/utils.go#L396-L403

Reply:
I don't think you can. The StatefulSet API doesn't have a way to return the pods. I believe the way to get them is by listing pods using the selector, as Arka did it (line 480).

Reply:
> I don't think you can. The StatefulSet API doesn't have a way to return the pods. I believe the way to get them is by listing pods using the selector, as Arka did it (line 480).

+1
I was wondering if certificate creation can be independent of pod creation/ready. If so, we can form the certificate/secret name `etcdcluster-sample-[n]` where `n=0, n< statefulset replicas`.